### PR TITLE
[10.0][FIX] prevent infinite loop in scheduler process

### PR DESCRIPTION
**Pull request by Cedric-Pigeon**
Description of the issue/feature this PR addresses:

In some cases the MRP scheduler can remain blocked on an infinite loop consuming a lot of CPU.
The reason comes from this loop: https://github.com/odoo/odoo/blob/10.0/addons/procurement/models/procurement.py#L293

If a new procument is created from another process (such like MTO orders) during this loop, the process never ends.

The only way to reproduce is in debug mode:

1/ Put a breakpoint on line https://github.com/odoo/odoo/blob/10.0/addons/procurement/models/procurement.py#L293

2/ Launch the MRP scheduler manually and stay paused on the breakpoint

3/ In parrallel, create and confirm a sale order with a MTO route, this will trigger the creation of a new procurement order in state '"Running"

4/ Continue debugging step by step the scheduler and you will notice that the loop never ends.

FYI, on our huge database, we encounter this problem on average once a week.
It took us a lot of time to be able to reproduce it as the use case is really tricky.

The goal of this PR in to prevent this behaviour

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by Cedric-Pigeon** [comment]
@jco-odoo do you have an opinion on this ?

**Comment by Cedric-Pigeon** [comment]
OPW-787651

File: addons/procurement/models/procurement.py [hunk 1]
```diff
@@ -290,11 +290,13 @@ def run_scheduler(self, use_new_cursor=False, company_id=False):
 
             # Check done procurements
             procurements = ProcurementSudo.search([('state', '=', 'running')] + (company_id and [('company_id', '=', company_id)] or []))
+            checked_procurements = list()
```
Comment (Elkasitu) [inline addons/procurement/models/procurement.py]: why not [] ?

Comment (Cedric-Pigeon) [reply addons/procurement/models/procurement.py]: no particular reason

Comment (Elkasitu) [reply addons/procurement/models/procurement.py]: please change it to `[]`, it's faster than `list()` and it's more pythonic

Comment (Cedric-Pigeon) [reply addons/procurement/models/procurement.py]: please, feel free to do it

**Comment by jco-odoo** [comment]
@Cedric-Pigeon a subtle one.  Your patch makes perfect sense.  A similar thing could be done with the confirmed to running/exception loop before, but it is not infinite as it will not stay in confirmed state.  Why not remove the loop all together for this last one?  The calculation is mostly redundant on itself as it will be set by the validation of the picking e.g.

**Comment by jco-odoo** [comment]
I proposed a fix in 10.0-scheduleralt-jco @sle-odoo

**Comment by jco-odoo** [comment]
@nim-odoo

**Comment by jco-odoo** [comment]
https://github.com/odoo-dev/odoo/commit/9f69d4148f7f4fc3238111af030eac4472e7890d

**Comment by Cedric-Pigeon** [comment]
@jco-odoo Thanks a lot for this quick and efficient feeback. :-)

**Comment by mreficent** [comment]
this should be closed...

**Pull request closed without merging**